platform/ios/src/MGLMapView.mm

```diff
@@ -5353,9 +5353,13 @@ - (void)insertCustomStyleLayerWithIdentifier:(NSString *)identifier preparationH
 {
     NSAssert(identifier, @"Style layer needs an identifier");
     MGLCustomStyleLayerHandlers *context = new MGLCustomStyleLayerHandlers(preparation, drawing, completion);
-    _mbglMap->addLayer(std::make_unique<mbgl::style::CustomLayer>(identifier.UTF8String, MGLPrepareCustomStyleLayer,
-                                                                  MGLDrawCustomStyleLayer, MGLFinishCustomStyleLayer, context),
-                       otherIdentifier ? mbgl::optional<std::string>(otherIdentifier.UTF8String) : mbgl::optional<std::string>());
+    try {
+        _mbglMap->addLayer(std::make_unique<mbgl::style::CustomLayer>(identifier.UTF8String, MGLPrepareCustomStyleLayer,
+                                                                      MGLDrawCustomStyleLayer, MGLFinishCustomStyleLayer, context),
+                           otherIdentifier ? mbgl::optional<std::string>(otherIdentifier.UTF8String) : mbgl::optional<std::string>());
+    } catch (std::runtime_error & err) {
+        [NSException raise:@"MGLRedundantLayerIdentiferException" format:@"%s", err.what()];
```

Comment:
If you’re in a good spot to do it, can you rename `MGLRedundantLayerIdentiferException` to `MGLRedundantLayerIdentiferException`? I don’t think we have any instances of it in the release branch, so this is a good branch to do it in. I can take care of `MGLRedundantSourceIdentiferException` separately: #7314.